library/x509_crt.c

```diff
@@ -1574,10 +1584,8 @@ int mbedtls_x509_crt_parse_path(mbedtls_x509_crt *chain, const char *path)
         if (file_data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) {
             continue;
         }
-
         w_ret = WideCharToMultiByte(CP_ACP, 0, file_data.cFileName,
-                                    -1,
-                                    p, (int) len,
+                                    -1, p, (int) len - 1,
```

Comment:
Why `len - 1`, not `len`? `WideCharToMultiByte()` takes into account the terminating null character and includes it as far as I can see. From the description of the input length from [the docs](https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-widechartomultibyte):
> `[in] cchWideChar`
> ...
> If this parameter is -1, the function processes the entire input string, including the terminating null character. Therefore, the resulting character string has a terminating null character, and the length returned by the function includes this character.

Which states that the output string will include the null-terminator when this function is passed a `-1` for the input length.

Reply:
Agreed, the original `len` should be kept here

Reply:
A bit further down 

> The lpMultiByteStr and lpWideCharStr pointers must not be the same. If they are the same, the function fails, and GetLastError returns ERROR_INVALID_PARAMETER.

We have set `cchWideChar = -1` so the described behaviour works on the `lpWideCharStr`. The len -1 is the size of the output buffer in bytes `lpMultiByteStr`

Tha being set it should work with len

~~~~~~~
    memset(filename, 0, MAX_PATH);
    memcpy(filename, path, len);
    filename[len++] = '\\';
    p = filename + len;
~~~~~~~

Were we allocate `MAX_PATH == 260` bytes and p is basically meant to be len width in bytes for this call as indicated by

`memset(p, 0, len);
`

I will change it and test it on the CI